### PR TITLE
v2.1.0 Updating EssentialsPE to latest API

### PR DESCRIPTION
**Pull request by JackMD**
Started working on updating the plugin to support PocketMine-MP latest API

### CODE REVIEW

File: src/EssentialsPE/EventHandlers/SignEvents.php [hunk 1]
```diff
@@ -164,7 +165,7 @@ public function onSignTap(PlayerInteractEvent $event): void{
                                 $item->setDamage(0);
                             }
                         }
-                        foreach ($event->getPlayer()->getInventory()->getArmorContents() as $item){
+                        foreach ($event->getPlayer()->getArmorInventory()->getContents() as $item){
                             if($this->getAPI()->isRepairable($item)){
                                 $item->setDamage(0);
```
Comment (dktapps) [inline src/EssentialsPE/EventHandlers/SignEvents.php]: this won't work... it's modifying a copy of the object, not the object in the inventory itself. I know this is not a bug you introduced, but be aware that this won't work as intended anyway.

File: src/EssentialsPE/Events/CreateAPIEvent.php [hunk 2]
```diff
@@ -14,10 +14,11 @@ class CreateAPIEvent extends PluginEvent{
...
 
-    /**
-     * @param Loader $plugin
-     * @param BaseAPI::class $api
-     */
+	/**
+	 * @param Loader $plugin
+	 * @param string $api
+	 * @internal param $BaseAPI ::class $api
+	 */
```
Comment (dktapps) [inline src/EssentialsPE/Events/CreateAPIEvent.php]: indentation is messed up here, and the final parameter noted there appears not to exist.

File: src/EssentialsPE/Events/CreateAPIEvent.php [hunk 3]
```diff
@@ -33,13 +34,14 @@ public function getClass(): string{
         return $this->class;
     }
 
-    /**
-     * @param BaseAPI::class $api
-     */
+	/**
+	 * @param string $class
+	 * @internal param $BaseAPI ::class $api
+	 */
```
Comment (dktapps) [inline src/EssentialsPE/Events/CreateAPIEvent.php]: same as above

File: src/EssentialsPE/Loader.php [hunk 4]
```diff
@@ -114,6 +113,7 @@ public function onEnable(): void{
             $this->getAPI()->fetchEssentialsPEUpdate(false);
         }
         $this->getAPI()->scheduleAutoAFKSetter();
+	$this->getLogger()->info("Plugin Enabled");
```
Comment (dktapps) [inline src/EssentialsPE/Loader.php]: seems unnecessary - also see indentation again

File: src/EssentialsPE/EventHandlers/SignEvents.php [hunk 5]
```diff
@@ -14,6 +14,7 @@
 use pocketmine\utils\TextFormat;
 
 class SignEvents extends BaseEventHandler{
+
```
Comment (dktapps) [inline src/EssentialsPE/EventHandlers/SignEvents.php]: unrelated whitespace change

File: src/EssentialsPE/Loader.php [hunk 6]
```diff
@@ -103,8 +103,7 @@ public function onEnable(): void{
         if(!is_dir($this->getDataFolder())){
             mkdir($this->getDataFolder());
         }
-        
-		$this->getLogger()->info(TextFormat::YELLOW . "Loading...");
```
Comment (dktapps) [inline src/EssentialsPE/Loader.php]: This change appears not to pertain to updating the API, nor does the one below.

File: src/EssentialsPE/EventHandlers/PlayerEvents.php [hunk 7]
```diff
@@ -17,12 +17,13 @@
...
 use pocketmine\event\player\PlayerQuitEvent;
-use pocketmine\event\TextContainer;
-use pocketmine\event\TranslationContainer;
+use pocketmine\lang\TextContainer;
+use pocketmine\lang\TranslationContainer;
 use pocketmine\Player;
 use pocketmine\utils\TextFormat;
 
 class PlayerEvents extends BaseEventHandler{
+
```
Comment (dktapps) [inline src/EssentialsPE/EventHandlers/PlayerEvents.php]: unnecessary whitespace change

File: src/EssentialsPE/Loader.php (hunk 4 above)
Comment (JackMD) [reply src/EssentialsPE/Loader.php]: Okay will remove it

File: src/EssentialsPE/Events/CreateAPIEvent.php (hunk 2 above)
Comment (JackMD) [reply src/EssentialsPE/Events/CreateAPIEvent.php]: Fixed

File: src/EssentialsPE/Events/CreateAPIEvent.php (hunk 3 above)
Comment (JackMD) [reply src/EssentialsPE/Events/CreateAPIEvent.php]: Fixed

File: src/EssentialsPE/Loader.php (hunk 4 above)
Comment (JackMD) [reply src/EssentialsPE/Loader.php]: Removed

File: src/EssentialsPE/Loader.php (hunk 6 above)
Comment (JackMD) [reply src/EssentialsPE/Loader.php]: I didn't get that quite right. It was removed as part of bit.ly/pmcolors

Comment (SOF3) [reply src/EssentialsPE/Loader.php]: I asked him to do this change.

File: src/EssentialsPE/Events/CreateAPIEvent.php [hunk 8]
```diff
@@ -14,32 +14,34 @@ class CreateAPIEvent extends PluginEvent{
...
-     * @param BaseAPI::class $api
-     */
-    public function __construct(Loader $plugin, string $api){
+	/**
+	 * CreateAPIEvent constructor.
+	 *
+	 * @param Loader $plugin
+	 * @param string $api
+	 */
+	public function __construct(Loader $plugin, string $api){
```
Comment (dktapps) [inline src/EssentialsPE/Events/CreateAPIEvent.php]: you broke the formatting more here instead of fixing it. The indentation should use spaces, not tabs.

File: src/EssentialsPE/Events/CreateAPIEvent.php [hunk 9]
```diff
@@ -14,32 +14,34 @@ class CreateAPIEvent extends PluginEvent{
...
     }
 
-    /**
-     * @return string
-     */
-    public function getClass(): string{
+	/**
+	 * @return string
+	 */
+	public function getClass(): string{
```
Comment (dktapps) [inline src/EssentialsPE/Events/CreateAPIEvent.php]: same here

File: src/EssentialsPE/Events/CreateAPIEvent.php [hunk 10]
```diff
@@ -14,32 +14,34 @@ class CreateAPIEvent extends PluginEvent{
...
     }
 
-    /**
-     * @param BaseAPI::class $api
-     */
-    public function setClass(string $class): void{
+	/**
+	 * @param string $class
+	 */
+	public function setClass(string $class): void{
```
Comment (dktapps) [inline src/EssentialsPE/Events/CreateAPIEvent.php]: same here

File: src/EssentialsPE/Events/CreateAPIEvent.php (hunk 8 above)
Comment (JackMD) [reply src/EssentialsPE/Events/CreateAPIEvent.php]: Updated PhpStorm settings

**Comment by JackMD** [comment]
Requesting for review. Changes have been tested and found to work.

**Comment by JackMD** [comment]
Uh wait. Found a bug.

**Comment by JackMD** [comment]
Seems good now

File: src/EssentialsPE/Commands/Repair.php [hunk 11]
```diff
@@ -48,24 +48,36 @@ public function execute(CommandSender $sender, string $alias, array $args): bool
             }
             foreach($sender->getInventory()->getContents() as $item){
                 if($this->getAPI()->isRepairable($item)){
-                    $item->setDamage(0);
+                    if($item->getDamage() > 0){
+                        $index = $sender->getInventory()->first($item, true);
```
[the remark this answers is not in the document]
Comment (JackMD) [reply src/EssentialsPE/Commands/Repair.php]: Then where would the `$item` come from? As of now the foreach loop gets the `$item` then the index of that very item item is found via `first()` and the item is set at that very index with damage zero.

[the remark this answers is not in the document]
Comment (JackMD) [reply src/EssentialsPE/Commands/Repair.php]: I have tested this by giving myself 15 diamond swords all with damage 100. I typed `/repair hand` and only the item I was holding was repaired and none other. Whereas executing `/repair all` command repaired everything I had in my inventory (armor inventory included).

[the remark this answers is not in the document]
Comment (JackMD) [reply src/EssentialsPE/Commands/Repair.php]: About the `foreach()` loop I will do that tomorrow

[the remark this answers is not in the document]
Comment (JackMD) [reply src/EssentialsPE/Commands/Repair.php]: @dktapps changes have been made

[the remark this answers is not in the document]
Comment (JackMD) [reply src/EssentialsPE/Commands/Repair.php]: Also just to add in I tested all of this again and it doesn't overwrite matching items in earlier slots.

**Comment by JackMD** [comment]
So is this PR accepted?

File: src/EssentialsPE/BaseFiles/BaseAPI.php [hunk 12]
```diff
@@ -428,9 +428,9 @@ public function getCurrencySymbol(): string{
     /**
      * Return the top 5 players with most money
      *
-     * @param CommandSender $player
+     * @param CommandSender|Player $player
      *
-     * @return bool
+     * @return bool|void
```
Comment (dktapps) [inline src/EssentialsPE/BaseFiles/BaseAPI.php]: this looks suspicious, it's typehinted as `bool`.

Comment (JackMD) [reply src/EssentialsPE/BaseFiles/BaseAPI.php]: Hmm looking at [this](https://github.com/poggit-orphanage/EssentialsPE/blob/fc3799185f756c27633e41c9b8161726c0f5ec8f/src/EssentialsPE/BaseFiles/BaseAPI.php#L435) its typehinted as `void`.

Comment (dktapps) [reply src/EssentialsPE/BaseFiles/BaseAPI.php]: since it's hinted as void, it's impossible for it to return bool.

**Comment by JackMD** [comment]
@dktapps please review

**Comment by edwinyoo44** [comment]
I think someone has updated to the new API
Try it:
https://github.com/iZeaoGamer/EssentialsPE

**Comment by dktapps** [comment]
that's a spoon of this repository, so it can't be merged into this repository.
